sdk/tables/azure-data-tables/tests/test_table_client.py

```diff
@@ -223,30 +224,262 @@ def test_client_with_sas_token(self, tables_storage_account_name, tables_primary
+        self, tables_storage_account_name, tables_primary_storage_account_key
+    ):
+        base_url = self.account_url(tables_storage_account_name, "table")
+        table_name = self.get_resource_name("mytable")
+        sas_token = self.generate_sas(
+            generate_account_sas,
+            tables_primary_storage_account_key,
+            resource_types=ResourceTypes.from_string("sco"),
+            permission=AccountSasPermissions.from_string("rwdlacu"),
+            expiry=datetime.utcnow() + timedelta(hours=1),
+        )
+        name_filter = "TableName eq '{}'".format(table_name)
 
-        sas_url = f"{base_url}/{table_name}?{sas_token}"
-        with TableClient.from_table_url(sas_url) as client:
+        with TableServiceClient(base_url, credential=AzureSasCredential(sas_token)) as client:
+            client.create_table(table_name)
+            result = client.query_tables(name_filter)
+            assert len(list(result)) == 1
+
+        conn_str = (
+            f"AccountName={tables_storage_account_name};SharedAccessSignature={AzureSasCredential(sas_token).signature}"
+        )
+        with TableServiceClient.from_connection_string(conn_str) as client:
+            result = client.query_tables(name_filter)
+            assert len(list(result)) == 1
+            client.delete_table(table_name)
+
+        with pytest.raises(ValueError) as ex:
+            TableServiceClient(f"{base_url}/?{sas_token}", credential=AzureSasCredential(sas_token))
+        ex_msg = "You cannot use AzureSasCredential when the resource URI also contains a Shared Access Signature."
+        assert ex_msg == str(ex.value)
+
+    @tables_decorator
+    @recorded_by_proxy
+    def test_table_client_with_token_credential(self, tables_storage_account_name, tables_primary_storage_account_key):
+        base_url = self.account_url(tables_storage_account_name, "table")
+        table_name = self.get_resource_name("mytable")
+        default_azure_credential = self.get_token_credential()
+        sas_token = self.generate_sas(
+            generate_account_sas,
+            tables_primary_storage_account_key,
+            resource_types=ResourceTypes.from_string("sco"),
+            permission=AccountSasPermissions.from_string("rwdlacu"),
+            expiry=datetime.utcnow() + timedelta(hours=1),
+        )
+
+        with TableClient(base_url, table_name, credential=default_azure_credential) as client:
+            table = client.create_table()
+            assert table.name == table_name
+
+        with TableClient.from_table_url(f"{base_url}/{table_name}", credential=default_azure_credential) as client:
+            entities = client.query_entities(
+                query_filter="PartitionKey eq @pk",
+                parameters={"pk": "dummy-pk"},
+            )
+            for e in entities:
+                pass
+
+        with TableClient(f"{base_url}/?{sas_token}", table_name, credential=default_azure_credential) as client:
```

Comment:
Same comment as above - we should add the `raw_request_hook` here to validate the auth being sent to the service.